modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSetOpTest.java

```diff
@@ -278,6 +278,25 @@ public void testSetOpDifferentNumericTypes(String setOp, int expectId, String ex
                 .check();
     }
 
+    @Test
+    public void testUnionDifferentNumericTypes() {
+        String query = "SELECT * FROM ("
+                + "SELECT id, val FROM t1 "
+                + "UNION "
+                + "SELECT id, val FROM t2) ORDER BY id";
+
+        assertQuery(query)
```

Comment:
JFYI, `assertQuery()..check()` by default tries to sort the "comparable" results until `ordered()` is set explicitly.